### PR TITLE
[baggage] Remove unused private field

### PR DESCRIPTION
**Pull request by yurishkuro**
There was no way to create a valid property with `hasData=false`, so the field is not necessary.

### CODE REVIEW

**Comment by MrAlias** [comment]
The following will create a `Property` with `hasData=false`:

```go
var p Property
```

**Comment by yurishkuro** [comment]
> var p Property

which would still be invalid because of empty key, and validity is the only thing `hasData` field impacts

File: baggage/baggage_test.go [hunk 1]
```diff
@@ -152,14 +152,11 @@ func TestNewKeyValueProperty(t *testing.T) {
 
 	p, err = NewKeyValueProperty("key", "value")
 	assert.NoError(t, err)
-	assert.Equal(t, Property{key: "key", value: "value", hasValue: true, hasData: true}, p)
+	assert.Equal(t, Property{key: "key", value: "value", hasValue: true}, p)
 }
 
 func TestPropertyValidate(t *testing.T) {
 	p := Property{}
-	assert.ErrorIs(t, p.validate(), errInvalidProperty)
```
Comment (pellared) [inline baggage/baggage_test.go:160]: I find `invalid baggage list-member property` a better error message than `invalid key`.

Comment (yurishkuro) [reply baggage/baggage_test.go:160]: It would be if this was baggage.validate(). Property.validate is a lower level, it doesn't know about baggage.

Comment (yurishkuro) [reply baggage/baggage_test.go:160]: That's not in scope of this PR. `member.validate` should be wrapping lower-level errors if it wants to return "invalid member" as top level error.

Comment (pellared) [reply baggage/baggage_test.go:160]: Gotcha 👍

**Review by pellared — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Pull request merged**